src/components/calcite-input/calcite-input.e2e.ts

```diff
@@ -1181,5 +1181,20 @@ describe("calcite-input", () => {
       expect(await calciteInput.getProperty("value")).toBe(initialValue);
       expect(await input.getProperty("value")).toBe(localizeNumberString(initialValue, "en-US", true));
     });
+
+    it("sets internals to readOnly or disabled when component is readOnly", async () => {
```

Comment:
Could this be rewritten to test the behavior and not the implementation details?